scripts/result_analysis.py

```diff
@@ -64,26 +198,32 @@ def plot_barchart(df, cat, min_year, max_year, class_dict, data):
     #     labels = [int(a) if a > 0 else "" for a in c.datavalues]
     #     ax.bar_label(c, label_type='center', color="black", labels=labels, fontsize=7)
 
-    plt.gca().set_yticks(plt.gca().get_yticks().tolist())
-    ticks_to_use = df.index[::5]
     labels = df[year][::5]
-
-    ax.set_xticks(ticks_to_use, labels, rotation=45, fontsize=10, ha='center')
-    ax.xaxis.set_minor_locator(MultipleLocator(1))
-    plt.xlabel('Year', fontweight='bold')
-
     title = class_dict[cat] if class_dict else cat
-    plt.title(title, fontweight='bold')
-    plt.legend(loc='upper left', frameon=False)    
-
-    plot_path = f'{data}_{title}.png'.replace(' ', '_')
-    plt.savefig(plot_path, bbox_inches='tight')
-    plt.close(fig)
+    plot_path = os.path.join(output_dir, f'{data}_{title}.png'.replace(' ', '_'))
+    format_barplot(df, ax, fig, labels, title, 'all_cantons', plot_path)
```

Comment:
L.204, error when calling the function:
` format_barplot(df, ax, fig, labels, title, plot_path)`